pkg/iptableswrapper/mocks/iptables_maps.go

```diff
@@ -113,6 +127,12 @@ func (ipt *MockIptables) ClearChain(table, chain string) error {
 }
 
 func (ipt *MockIptables) DeleteChain(table, chain string) error {
+	// More than just the create chain rule
+	if len(ipt.DataplaneState[table][chain]) > 1 {
+		err := fmt.Sprintf("Chain %s is not empty", chain)
```

Comment:
Why wouldn't we delete the chain if it has rules in it?

Reply:
To mock the behavior from the actual function https://pkg.go.dev/github.com/coreos/go-iptables/iptables#IPTables.DeleteChain